free-1.md

```diff
@@ -57,12 +98,30 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+I would guess that quite a bit of time would go into figuring out exactly how
+the user should interact with the language. This language would be some
+combination of formal logic, general-purpose programming language, and English.
+It needs to be easy to use but also extremely general --- it would be bad if a
+game _almost_ fits in the language because then the language becomes useless.
+Programming the interpreter for the language would also be difficult but
+probably not as bad. If the language is closer to formal logic, some basic
+translation could happen and then be fed into a logic prover. Outputting game
+rules would be complex if one wanted them to look good but simple to get
+something basic done.
```

Comment:
I think a possible solution here is allowing some amount of "nebulousness". If there's a game state that is hard to describe, just take it as a primitive. The tradeoff is that the checker will not be able to know anything about how one might get into that state, 

However, if the checker IS sophisticated enough to be making complicated checks on things like "this rule is fine because this other state could never happen given ..." then it's just as powerful as a theorem prover.

But, if the difference is just what people express as black boxes or not, I think the general case of your idea would *also* be as strong as a theorem prover, and moreover as a game *implementation*.  Which means it might be better for it to work more like proof assistants do.

Actually, I think proof assistants in general are a very similar thing as this project is and would be worth looking into.

Incidentally, if you actually want to use formal logic, but perhaps are looking for a reformulation that is more similar to how people actually express their ideas, you could check out Discourse Representation Theory. Probably want to avoid those things as much as possible though.